src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -612,7 +612,7 @@ ActStatus ActionHandlers::send_esm_info_req_to_ue(SM::ControlBlock& cb)
 	esmreq.dl_seq_no = ue_ctxt->getDwnLnkSeqNo();
 	memcpy(&(esmreq.int_key), &((ue_ctxt->getUeSecInfo().secinfo_m).int_key),
 			NAS_INT_KEY_SIZE);
-	ue_ctxt->setDwnLnkSeqNo(esmreq.dl_seq_no+1);
+	ue_ctxt->incrementDwnLnkSeqNo();
```

Comment:
@badhrinathpa This API may not be good right ? Given that sequence number is not just counter but more than that.

Reply:
I have added the right way to update sequence numbers and counters in the security branch. 
We can keep the metrics generator code part though. that will be useful.

Reply:
Ok. Retained the script changes. Reverted the Action handler changes.
Please review https://github.com/omec-project/Nucleus/pull/37